### PR TITLE
fix: handle whitespace according to JSX common practice

### LINKED ISSUE #5849: Wrong code generated for JSX syntax
### Rollup Version

4.34.8

### Operating System (or Browser)

NA

### Node Version (if applicable)

NA

### Link To Reproduction

https://rollupjs.org/repl/?version=4.34.8&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZnVuY3Rpb24lMjBmKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTNDZGl2JTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUzQ2ltZyUyMCUyRiUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlMkZkaXYlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTNDJTJGJTNFJTVDbiUyMCUyMCUyMCUyMCklM0IlNUNuJTdEJTVDbiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwZiUzQiU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmpzeCUyMiUzQXRydWUlMkMlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiU3RCU3RCU3RA==

### Expected Behaviour

Having in rollup config:
```
const rollupOptions = {
    ...
    jsx: {
        mode: 'classic',
        factory: 'createElement',
        fragment: 'createFragment',
        importSource: null,
    },
};

```
And input file:
```
    return (
        <>
            <div>
                <img />
            </div>
        </>
    );

```

JSX syntax assumes that all whitespaces at the beginning and end of each line (and empty lines, and newline characters after tag) should be removed.

Expected output (split into lines for readability):
```
    return (
        /*#__PURE__*/createElement(createFragment, null,
            /*#__PURE__*/createElement("div", null,
                /*#__PURE__*/createElement("img", null)
            )
        )
    );

```

### Actual Behaviour

Rollup generates (split into lines for readability):
```
    return (
        /*#__PURE__*/createElement(createFragment, null,
            "\n            ",
            /*#__PURE__*/createElement("div", null,
                "\n                ",
                /*#__PURE__*/createElement("img", null),
                "\n            "
            ),
            "\n        "
        )
    );

```

### PR DESCRIPTION
**Pull request by cyyynthia**
This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers:

- Resolves #5849 

### Description

This PR fixes the handling of whitespace within JSX text, matching the *generally agreed* (for the lack of a proper spec) upon behavior of whitespace in React.

It also fixes an edge-case where invalid JS code was generated if a plaintext attribute had a newline in it. Those do not obey to the same whitespace rules as text.

#### References
- https://github.com/facebook/react/pull/480#issuecomment-31296039
- https://github.com/facebook/react/issues/896

### CODE REVIEW

File: src/ast/nodes/JSXText.ts [hunk 1]
```diff
@@ -3,19 +3,36 @@ import type { NormalizedJsxOptions } from '../../rollup/types';
 import type * as NodeType from './NodeType';
 import { NodeBase, onlyIncludeSelf } from './shared/Node';
 
+const RE_WHITESPACE_TRIM = /^[ \t]*\r?\n[ \t]*|[ \t]*\r?\n[ \t]*$/g;
```
Comment (lukastaegert) [inline src/ast/nodes/JSXText.ts]: This looks like it will only remove until the first line-break, but checking against the Babel REPL, multiple line-breaks are removed as well. Compare this:

Babel: https://babeljs.io/repl#?config_lz=N4IgZglgNgpgdgQwLYxALhAJxgBygOgCsBnADxABoQdtiYAXY9AbWZHgDdLR6FMBzBkwwATGGAQBXKIwoACOAHt6ciDDkBGDfKUq1AfSSKARpo2UQRkdJjCJUOlWOT-kUrfT1MkmAF8AuhRs2AgAxvTcWJJw9BAo6CBS9IpICLGhIAFBIMS8ggC0AEyRYqGKmGnlxABqMJjEEIpwCYUADIUAzPlaFjgQODBQEHAwAAqYijiKxAhQCQAWYQDWmf6BOYqSmKEwACoAngMJVjaZQA&code_lz=KYDwDg9gTgLgBAMwK4DsDGMCWEWIBQCUcA3gFBwVxTAxJS57mXMA8AfE8851xT7yH5cWAE0wA3APQdeFFtJ4EA3KQC-pIA&lineWrap=true&version=7.28.3

Rollup: https://rollup-90iirwpg8-rollup-js.vercel.app/repl/?pr=6051&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBmKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjB4JTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUzQ2RpdiUyRiUzRSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlMkYlM0UlNUNuJTIwJTIwJTIwJTIwKSUzQiU1Q24lN0QlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJqc3glMjIlM0ElMjJyZWFjdC1qc3glMjIlMkMlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiU3RCU3RCU3RA==

Note the ominous double empty space `"  x"` it creates, which is from the second replacement function.

Comment (cyyynthia) [reply src/ast/nodes/JSXText.ts]: Ah, good catch. I updated the regex and added tests for these cases

**Review by lukastaegert — COMMENTED** [review]
Thanks! I just stumbled across another issue with the tree-shaking of white-space text. Compare this https://rollup-90iirwpg8-rollup-js.vercel.app/repl/?pr=6051&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBmKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlM0UlM0NkaXYlMkYlM0UlM0MlMkYlM0UlNUNuJTIwJTIwJTIwJTIwKSUzQiU1Q24lN0QlNUNuJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlMkMlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJqc3glMjIlM0ElMjJyZWFjdC1qc3glMjIlMkMlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiU3RCU3RCU3RA== with this https://rollup-90iirwpg8-rollup-js.vercel.app/repl/?pr=6051&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQSUyMiUyMiUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBmKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwKCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlM0MlM0UlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTNDZGl2JTJGJTNFJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUzQyUyRiUzRSU1Q24lMjAlMjAlMjAlMjApJTNCJTVDbiU3RCU1Q24lMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmpzeCUyMiUzQSUyMnJlYWN0LWpzeCUyMiUyQyUyMm91dHB1dCUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTdEJTdEJTdE

If a text is completely removed because the text only contains a line-break with spaces, then it might use the `jsxs` function even though it should rather switch to the `jsx` function. In the example, it is particularly problematic because it uses the `jsxs` function without an array.

So the logic to detect whether to use an array correctly detects this case, but the logic to decide whether to use `jsxs/jsx` does not...

**Comment by cyyynthia** [comment]
Ah! You're good with coming up with these edge cases, I should've been a bit more careful when implementing it 😄

Adjusted the logic and added a test case 👍🏻
